lib/progress.js

```diff
@@ -137,7 +137,21 @@ export function estimate(requestAsync, isBrowser) {
 				finishFlush: zlib.constants.Z_SYNC_FLUSH,
 			};
 
+			// Allow overriding this behaviour by setting the ZLIB_FLUSH env var
+			// to one of the allowed zlib flush values (process.env.ZLIB_FLUSH="Z_NO_FLUSH").
+			// https://github.com/nodejs/node/blob/master/doc/api/zlib.md#zlib-constants
+			if (
+				process.env.ZLIB_FLUSH &&
+				zlib.constants[process.env.ZLIB_FLUSH] > -1
```

Comment:
Just wondering about the `> -1` part. Was the intention to check whether `process.env.ZLIB_FLUSH` has the name of a valid zlib constant? I.e. to guard against `process.env.ZLIB_FLUSH='foo'` or such. I gather that the constant values are:
```
#define Z_NO_FLUSH      0
#define Z_PARTIAL_FLUSH 1
#define Z_SYNC_FLUSH    2
#define Z_FULL_FLUSH    3
#define Z_FINISH        4
#define Z_BLOCK         5
#define Z_TREES         6
```
and perhaps that's where `> -1` comes from, but it relies on the fact that `undefined > -1` is false, which I find a bit uncomfortable because:
```
$ node
> const zlib = require('zlib');
> zlib.constants['foo']
undefined
> undefined > -1
false
> undefined < -1
false
> undefined == -1
false
> undefined === -1
false
```

I think it would it would be clearer to more directly test whether `process.env.ZLIB_FLUSH` defines a valid constant name with:
```
	if (
		process.env.ZLIB_FLUSH &&
		process.env.ZLIB_FLUSH in zlib.constants
```
or
```
	if (
		process.env.ZLIB_FLUSH &&
		zlib.constants[process.env.ZLIB_FLUSH] != null
```
given that:
```
$ node
> const zlib = require('zlib');
> zlib.constants['Z_NO_FLUSH']
0
> 'Z_NO_FLUSH' in zlib.constants
true
> 'foo' in zlib.constants
false
> '' in zlib.constants
false
> undefined in zlib.constants
false
> zlib.constants['Z_NO_FLUSH'] != null
true
> zlib.constants[''] != null
false
> zlib.constants[undefined] != null
false
> undefined != null
false
> 0 != null
true
```

Reply:
```nodejs
	if (
		process.env.ZLIB_FLUSH &&
		process.env.ZLIB_FLUSH in zlib.constants
```
@pdcastro This is exactly what I was trying to accomplish but failed to figure out how to do it properly. Thanks for showing me the way!